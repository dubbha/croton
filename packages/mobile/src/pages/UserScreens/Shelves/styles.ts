import { StyleSheet } from 'react-native';
import { COMPONENTS_STYLE, COLORS } from '../../../styles';

export default StyleSheet.create({
  shelves: {
    flex: 1,
    flexDirection: 'column',
    paddingHorizontal: COMPONENTS_STYLE.mainPaddingHorizontal,
  },

  shelves__header: {
    flexDirection: 'row',
    justifyContent: 'center',
    marginTop: 20,
  },

  shelves__body: {
    flex: 1,
  },

  shelves__modal: {
    flex: 1,
  },

  shelves__modal__body: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    backgroundColor: COLORS.lightMain,
  },

  shelves__listWrap: {
    flex: 1,
    paddingVertical: 10,
  },

  shelves__list: {
    flex: 1,
  },

  shelves__item: {
    flex: 0.3,
    marginTop: 10,
    minHeight: 50,
  },

  shelves__message: {
    minHeight: 30,
    marginVertical: 10,
  },
});
