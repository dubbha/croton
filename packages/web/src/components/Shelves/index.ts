export * from './Shelves';
export * from './Shelf';
export * from './Flower';
