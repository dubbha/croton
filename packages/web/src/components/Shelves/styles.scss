@import 'assets/styles/theme.scss';

.shelves {
  &.container {
    justify-content: start;
    max-width: 80%;
    margin-bottom: 50px;
  }

  &-empty {
    margin: 30px auto;
    min-width: 70%;

    form {
      max-width: 100%;
    }
  }

  &-header {
    color: $color-grass-dark;
  }
}

.shelf-item,
.flower-item {
  margin: 10px auto;
  display: flex;
  flex-direction: column;
  width: 30%;
  padding: 10px;
  min-height: 140px;

  &-new {
    display: flex;
    align-items: center;
    justify-content: center;
    min-height: 100%;
    min-width: 100%;
    color: $color-grass;

    &:hover {
      color: $color-grass-dark;
    }
  }
}
