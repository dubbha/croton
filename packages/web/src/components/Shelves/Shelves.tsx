import React from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSeedling } from '@fortawesome/free-solid-svg-icons';
import { ShelfList } from './ShelfList';
import './styles.scss';

export const Shelves = () => {
  // mock data
  const shelves = [
    {
      id: 1,
      name: 'qweqe',
      location: 'some location',
      description: 'description'
    },
    {
      id: 2,
      name: 'My awesome Shelf',
      location: 'Super Location',
      description: 'Description of my awesome shelf'
    },
    {
      id: 3,
      name: 'qweqe',
      location: 'some location',
      description: 'description'
    },
    {
      id: 4,
      name: 'My awesome Shelf',
      location: 'Super Location',
      description: 'Description of my awesome shelfDescription of my awesome shelfDescription of my awesome shelfDescription of my awesome shelfDescription of my awesome shelfDescription of my awesome shelf'
    },
    {
      id: 5,
      name: 'qweqe',
      location: 'some location',
      description: 'description'
    },
    {
      id: 6,
      name: 'My awesome Shelf',
      location: 'Super Location',
      description: 'Description of my awesome shelf'
    },
    {
      id: 7,
      name: 'qweqe',
      location: 'some location',
      description: 'description'
    },
    {
      id: 8,
      name: 'My awesome Shelf',
      location: 'Super Location',
      description: 'Description of my awesome shelf'
    },
    {
      id: 9,
      name: 'qweqe',
      location: 'some location',
      description: 'description'
    },
    {
      id: 10,
      name: 'My awesome Shelf',
      location: 'Super Location',
      description: 'Description of my awesome shelfDescription of my awesome shelfDescription of my awesome shelfDescription of my awesome shelf'
    },
  ];
  // end mock data

  return (
    <div className="container shelves">
      <h1 className="shelves-header">
        <FontAwesomeIcon icon={faSeedling} />
        <span className="icon-prefix">My Shelves</span>
      </h1>
      { shelves.length
        ? <ShelfList shelves={shelves} />
        : (
          <div className="shelves-empty">
            <h4>You Don&#39;t Have Any Shelves Yet</h4>
            <Link to="/profile/add-shelf">Create Shelf</Link>
          </div>
        )}
    </div>
  );
};
