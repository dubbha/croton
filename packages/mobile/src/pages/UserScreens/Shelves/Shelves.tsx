import React, { useState, useEffect } from 'react';
import {
  SafeAreaView,
  View,
  Text,
  Modal,
  TouchableOpacity,
} from 'react-native';
import { useDispatch, useSelector } from 'react-redux';

import styles from './styles';
import { SCREEN_USER_SHELF } from '../../screens';
import { SHELVES_GET } from '../../../store/shelves/actions';
import { ShelfInterface } from '../../../components/Shelf/interface';
import { Shelf } from '../../../components/Shelf';
import { ShelfFormConfig } from '../../../components/ShelfForms';
import { CustomButton } from '../../../components/Button';
import { FlatList } from 'react-native-gesture-handler';

export const Shelves = ({ navigation }: any) => {
  const [isShowShelfFormConfig, setIsShowShelfFormConfig] = useState(false);
  const dispatch = useDispatch();
  const shelves: ShelfInterface[] = useSelector(state => {
    // TODO: main property shelves should rename, maybe in 'products'
    return state.shelves.shelves;
  });

  useEffect(() => {
    dispatch({ type: SHELVES_GET });
  }, [dispatch]);

  const renderShelfSettings = () => (
    <Modal animationType="fade" transparent={true}>
      <View style={styles.shelves__modal__body}>
        <ShelfFormConfig
          closeFunc={() => {
            dispatch({ type: SHELVES_GET });
            setIsShowShelfFormConfig(false);
          }}
        />
      </View>
    </Modal>
  );

  const renderShelf = ({ item: shelf, index }: any) => {
    return (
      <TouchableOpacity
        onPress={() => navigation.navigate(SCREEN_USER_SHELF, { ...shelf })}
        style={[
          styles.shelves__item,
          index === 0 ? styles.shelves__item__first : undefined,
        ]}>
        <Shelf {...shelf} />
      </TouchableOpacity>
    );
  };

  const renderShelves = () => {
    if (!shelves || !shelves.length) {
      return (
        <View>
          <Text>Shelves are empty</Text>
        </View>
      );
    } else {
      return (
        <FlatList
          data={shelves}
          keyExtractor={item => item.id.toString()}
          renderItem={renderShelf}
        />
      );
    }
  };

  return (
    <SafeAreaView style={styles.shelves}>
      <View style={styles.shelves__content}>
        <View style={styles.shelves__body}>
          <View style={styles.shelves__listWrap}>
            {shelves && renderShelves()}
          </View>
        </View>
        <View style={styles.shelves__footer}>
          <CustomButton
            title="Add shelf"
            onPress={() => {
              setIsShowShelfFormConfig(true);
            }}
          />
        </View>
        {isShowShelfFormConfig && renderShelfSettings()}
      </View>
    </SafeAreaView>
  );
};
