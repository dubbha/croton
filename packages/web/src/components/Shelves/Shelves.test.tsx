import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Shelves } from './Shelves';

jest.mock('./ShelfList', () => ({
  ShelfList: () => <div>Shelf List</div>,
}));

describe('components/Shelves', () => {
  it('should render successfully', () => {
    const { container } = render(<Shelves />, { wrapper: MemoryRouter });

    expect(container.firstChild).toMatchSnapshot();
  });
});
