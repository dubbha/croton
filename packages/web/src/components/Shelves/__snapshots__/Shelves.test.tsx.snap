// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Shelves should render successfully 1`] = `
<div
  class="container shelves"
>
  <h1
    class="shelves-header"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-seedling fa-w-16 "
      data-icon="seedling"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M64 96H0c0 123.7 100.3 224 224 224v144c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16V320C288 196.3 187.7 96 64 96zm384-64c-84.2 0-157.4 46.5-195.7 115.2 27.7 30.2 48.2 66.9 59 107.6C424 243.1 512 147.9 512 32h-64z"
        fill="currentColor"
      />
    </svg>
    <span
      class="icon-prefix"
    >
      My Shelves
    </span>
  </h1>
  <div>
    Shelf List
  </div>
</div>
`;
